cogs/fun.py

```diff
@@ -425,42 +425,48 @@ async def bill(self, ctx):
 	@commands.command(hidden=True)
 	@commands.cooldown(rate=3, per=10.0, type=commands.BucketType.user)
 	@commands.bot_has_permissions(embed_links=True)
-	async def xkcd(self, ctx, *, id: int = None):
+	async def xkcd(self, ctx, id: typing.Union[int, str] = None):
 		'''Get a random or specified xkcd comic.'''
-
-		if id is None:
-			url = 'https://c.xkcd.com/random/comic/'
-		else:
-			url = 'https://xkcd.com/{}'.format(id)
-
 		async with ctx.typing():
-			async with ctx.http.get(url) as resp:
+			resp = None
+			comic_json = None
+			url = None
+			if isinstance(id,int) and not id == 0:
+				# specific comic
+				id = abs(id)
+				url = f"https://xkcd.com/{id}/info.0.json"
```

Comment:
```suggestion
			if isinstance(id, int) and id > 0:
				# specific comic
				url = 'https://xkcd.com/{}/info.0.json'.format(id)
```
Negatives here don't make sense imo.
Also keep consistency:
* single quotes
* spaces after commas
* `''.format` rather than `f''` for compatibility (f-strings were introduced in Python 3.6)